.github/workflows/snapshots.yml

```diff
@@ -0,0 +1,31 @@
+name: Update Snapshots
+
+on:
+  workflow_dispatch:
+
+jobs:
+  update_snapshots:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Fail if branch is main
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          echo "This workflow should not be triggered with workflow_dispatch on main"
+          exit 1
+
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm ci
+      - name: Install Playwright Browsers
+        run: npx playwright install --with-deps chromium
+      - name: Run Playwright tests
+        run: npm run e2e:snapshots
+      - name: Commit Playwright updated snapshots
+        uses: EndBug/add-and-commit@v9
+        # We don't want to commit formatting changes to main
+        if: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
Can't you configure the whole workflow to not run on main?
Or why is this `if` only here and not at the top?

Reply:
I don't think you can configure `workflow_dispatch` to not be available on `main`.

That's why I added a guard at the start of this job.

```
      - name: Fail if branch is main
        if: ${{ github.ref == 'refs/heads/main' }}
        run: |
          echo "This workflow should not be triggered with workflow_dispatch on main"
          exit 1
```

But also added the additional `if` line you commented "just in case" and also because we use `EndBug/add-and-commit` in NNS-dapp and II always with a `if` - to follow the same pattern.

Reply:
If you know how to configure `workflow_dispatch` to not be available on main, of course happy to hear it!